File: Comparative-Analysis/fibonacci.go
package main

import (
	"fmt"
	"time"
)

func fib(n int) int {
	if n <= 2 {
		return 1
	}
	return fib(n-1) + fib(n-2)
}
func main() {
	start := time.Now()
	fmt.Println(fib(30))
	fmt.Println("Time:", time.Since(start))
}

File: Comparative-Analysis/fibonacci.cpp
#include <iostream>
#include <chrono>
using namespace std;

int fib(int n) {
    return n <= 2 ? 1 : fib(n - 1) + fib(n - 2);
}

int main() {
    auto start = chrono::high_resolution_clock::now();
    cout << fib(30) << endl;
    auto end = chrono::high_resolution_clock::now();
    cout << "Time: " << chrono::duration<double>(end - start).count() << "s\n";
}

